### ISSUE TITLE
Deprecation of the Grid element

### ISSUE THREAD

**Issue by MarcinZiabek** [QuestPDF/QuestPDF#380]
## What is the Grid element?

The Grid element introduces a new type of layout that equally divides available space into columns of equal width. Elements may take one or more columns. It is possible to put space between columns. This algorithm is well known from design systems such as [bootstrap](https://getbootstrap.com/docs/4.0/layout/grid/).

Read more in [the official documentation](https://www.questpdf.com/api-reference/grid).

## Purpose

Historically, the Grid element has been implemented to simplify development workflow. It allows an easy creation of [more complex layout structures](https://www.questpdf.com/examples/complex-layouts) with minimized nesting. 

Important: everything tha can be accomplished with the Grid element, can also be implemented using the Column and the Row elements.

## Internal implementation

Internally, the Grid element is implementedd using the component approach. All items are divided into lines and then represented using the combination of the Column and the Row elements.

To achieve alignment (left/center/right), proper empty elements are injected before and/or after the content within a line.

Read [the actual implementation code](https://github.com/QuestPDF/QuestPDF/blob/main/QuestPDF/Elements/Grid.cs).

## Problems

While implementing the right-to-left content direction support, I have experienced problems with the Grid element content alignment. Because the alignment is achieved by putting empty content before / after the actual children, the right-to-left implementation will always flip the alignment (right alignment will become left alignment).

To achieve proper results, I may need to reimplement the Grid element and stop using the component approach with pre-calculated spacing.

## Proposition

The Grid element does not introduce any significant value to the library. Let's deprecate this element.

## Alternatives

As mentioned before, the Grid element has been introduced to simplify workflow with more advanced layout structures. [The Table element](https://www.questpdf.com/api-reference/table) can be considered as even more powerful and flexible approach.

**Comment by MarcinZiabek** [comment]
@girlpunk @Bebo-Maker I would love to know your opinion on this topic 😁

**Comment by girlpunk** [comment]
I think combinations of column/row elements, or a table element, should reasonably be able to recreate anything using grid at the moment. However, there are certainly some situations where more advanced functionality would need to be recreated manually (such as a CSS Grid style layout with an arbitrary or runtime-defined number and size of elements)

**Comment by MarcinZiabek** [comment]
> However, there are certainly some situations where more advanced functionality would need to be recreated manually (such as a CSS Grid style layout with an arbitrary or runtime-defined number and size of elements)

As far as I see, the current QuestPDF Table implementation closly resembles features provided by the CSS Grid. Of course, there are some differences, mainly due to environments where those algorithms operate, limited page size vs infinite in size webpage.

On the other hand, QuestPDF Grid algorithm is much closer to the Bootstrap Grid, that was used to structure content way before CSS Grid has been introduced in CSS. 

And as such, QuestPDF Grid does not present any significant value in my opinion. It can be replaced by either the Table element or the combination of the Column and the Row elements.

If any, we may want to consider adding spacing capability to the Table algorithm, so to make it even more useful.

**Comment by Bebo-Maker** [comment]
Sorry for the late response. Can only agree with the points already stated. Pretty much every use case can be handled by the `Table` element or a combination of `Row` and `Column` elements.

**Comment by MarcinZiabek** [comment]
The decision has been made. The `Grid` element has been deprecated in favor of the `Row`, `Column` and `Table` elements.

**Comment by aaronwjha** [comment]
Might be a good idea to update the docs to reflect this decision.

**Comment by MarcinZiabek** [comment]
I agree! 😀 I am considering simply removing the documentation for deprecated elements. This deprecation is already quite old. What do you think?

**Comment by aaronwjha** [comment]
Yes, probably the most logical decision. For Grid, I suggest keeping a page showing how to accomplish the layout using tables or columns/rows.

**Comment by TheAustinG** [comment]
@MarcinZiabek Just stopping by to say that I think it would be very beneficial to at least mention in the Grid docs (which still exist), that Grid is now deprecated. I only found out when I tried to use Grid and intellisense alerted me that it has been deprecated. I had to come here and search the issues to confirm that it truly is deprecated and shouldn't be used anymore.

It would also be great to show an example of how to use a table or rows & columns to mimic the current Grid example.

Love everything else about QuestPDF!
